### PR TITLE
Refactor util packages into pkg/util

### PR DESCRIPTION
**Pull request by rfay**
## The Problem:

I wanted to add one simple utility function (CheckErr()) and we had no generic utility package to put it in.

## The Fix:

Refactor all of pkg/local/utils.go and some of pkg/platform/utils.go into pkg/util, which is where @tannerjfco has up-and-coming utility files.

There was at least one function that was unused so removed.
There were a couple of functions that were copy-pasted into two utils-related functions, so one was removed.

## The Test:

This should not change any test results or coverage.

## Release/Deployment notes:

Since my experience with ddev and its codebase is less than @beeradb and @tannerjfco I'm sure there will be some things that need to be changed about this.

### CODE REVIEW

File: pkg/util/utils.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// FormatPlural is a simple wrapper which returns different strings based on the count value.
+func FormatPlural(count int, single string, plural string) string {
+	if count == 1 {
+		return single
+	}
+	return plural
+}
+
+// RenderAppTable will format a table for user display based on a list of apps.
+func RenderAppTable(apps map[string]map[string]string, name string) {
```
Comment (beeradb) [inline pkg/util/utils.go]: Not sure this should actually be in utils. As far as I know it's not used outside of the platform plugins, and doesn't feel very "util" like. to me.

Comment (rfay) [inline pkg/util/utils.go]: It can move back. I wondered also.

Comment (tannerjfco) [inline pkg/util/utils.go]: It should probably be moved to its own package, similar to appimport and ddevapp for config. I believe list functionality is intended to be the same across platforms.

Comment (rfay) [inline pkg/util/utils.go]: Maybe RenderAppTable() *could* live in util at this point then? In general, it seems like we've erred on the side of being too specific in placement (and then sometimes copy/pasting, which is evil :) )

Comment (beeradb) [inline pkg/util/utils.go]: `A little copying is better than a little dependency.` - Rob Pike (https://go-proverbs.github.io/) 

:)

Comment (beeradb) [inline pkg/util/utils.go]: FYI I mostly said that to be cute, not to derail the conversation about whether this should be in utils or not.

**Comment by rfay** [comment]
refactor here was rebased after the import PR.

**Pull request merged**